src/main/java/org/openeo/spring/api/JobsApiController.java

```diff
@@ -379,78 +429,222 @@ public ResponseEntity<?> debugJob(
 		LogEntries logEntries = new LogEntries();
-		//TODO describe query
-		String elasticSearchQuery = "filebeat-7.13.3-2021.07.13-000001/_search";
+		
+		// Construcing ES query URL on selected index
+		String elasticSearchQuery = serviceIndexName + "/_search?size=10000"; //TODO (Lorenzo): sostituire
+		// il size con il metodo di paging indicato dalla documentazione (serve però anche, per facilitare 
+		// questo, riscrivere la query sotto direttamente tramite oggetto JSON anziché StringBuilder)
+				
 		try {
-			//TODO query elastic search endpoint here for all log information regarding a job queued for processing.
-			URL url = new URL(elasticSearchEndpoint + "/" + elasticSearchQuery);
-			HttpURLConnection conn = (HttpURLConnection) url.openConnection();
-			conn.setRequestMethod("GET");
-			conn.setDoOutput(true);
+			
+	        HttpURLConnection conn = null;
+			URL url = new URL(elasticSearchEndpoint + elasticSearchQuery);
+			int check_code = HttpURLConnection.HTTP_OK;
+			
+			//check if elasticSearchEndpoint is an HTTP or HTTPS URL
+			if (elasticSearchEndpoint.startsWith("https://")) // break point here (10 aug 23)
+			{
+				HostnameVerifier customHostnameVerifier = new HostnameVerifier() {
+				    @Override
+				    public boolean verify(String hostname, SSLSession session) {
+				        // In questa implementazione, considereremo qualsiasi nome di host valido.
+				        // Se hai esigenze specifiche di verifica, puoi aggiungere la logica qui.
+				        return true;
+				    }
+				};
+
+				// Imposto come default questo verificatore personalizzato HTTPS temporaneo (TOFIX)
+				HttpsURLConnection.setDefaultHostnameVerifier(customHostnameVerifier);
+				
+				// Load the truststore
+				char[] trustStorePassword_ = trustStorePassword.toCharArray();
+				KeyStore trustStore = KeyStore.getInstance("JKS");
+				try (InputStream trustStoreInputStream = new FileInputStream(trustStorePath)) {
+					trustStore.load(trustStoreInputStream, trustStorePassword_);
+				}
+		
+				// Create TrustManagerFactory
+				TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+				trustManagerFactory.init(trustStore);
+		
+				// Load the keystore
+				char[] keyStorePassword_ = keyStorePassword.toCharArray();
+				KeyStore keyStore = KeyStore.getInstance("JKS");
+				try (InputStream keyStoreInputStream = new FileInputStream(keyStorePath)) {
+					keyStore.load(keyStoreInputStream, keyStorePassword_);
+				}
+		
+				// Create KeyManagerFactory
+				KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+				keyManagerFactory.init(keyStore, keyStorePassword_);
+		
+				// Create SSLContext
+				SSLContext sslContext = SSLContext.getInstance("TLS");
+				sslContext.init(keyManagerFactory.getKeyManagers(), trustManagerFactory.getTrustManagers(), null);
```

Comment:
Do we need all this part? [**`HttpsUrlConnection`**](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/javax/net/ssl/HttpsURLConnection.html) already provides its default verifier and SSL factory. These are also actions to be probably done once, not per-request, maybe (this might affect the performance)

Reply:
Post-poned